sbi/utils/sbiutils.py

```diff
@@ -1006,3 +1018,43 @@ def seed_all_backends(seed: Optional[Union[int, Tensor]] = None) -> None:
     torch.cuda.manual_seed(seed)
     torch.backends.cudnn.deterministic = True  # type: ignore
     torch.backends.cudnn.benchmark = False  # type: ignore
+
+
+def warn_if_deprecated(
+    method: Callable, locals_dict: Dict[str, Any], deprecated_keys: Set
+) -> None:
+    """
+    Issues a warning if any deprecated parameters are used with non-default values.
+
+    This function compares the values of deprecated parameters (from `locals_dict`)
+    against their default values in the given `method` signature. If a deprecated
+    parameter is explicitly set to a non-default value, a `DeprecationWarning` is
+    raised.
+
+    Args:
+        method: The function whose parameters are checked.
+        locals_dict: The arguments of the function.
+        deprecated_keys: The names of the parameters that are deprecated.
+
+    """
+
+    # Get the signature of the function
+    method_signature = inspect.signature(method)
+
+    used = []
+    for key in deprecated_keys:
+        if key in locals_dict and key in method_signature.parameters:
+            default_value = method_signature.parameters[key].default
+
+            # Compare value to default
+            if locals_dict[key] != default_value:
+                used.append(key)
+
+    if used:
+        warnings.warn(
+            f"The following arguments are deprecated and"
+            " will be removed in a future version: "
+            f"{', '.join(used)}. Please use `posterior_parameters` instead.",
```

Comment:
We could refer to some documentation of how to use this new API, e.g., to the key PR, or to another tutorial where we show it, e.g, the sampling interface tutorial, or add it to a "how-to-guide" like here: https://sbi.readthedocs.io/en/latest/how_to_guide/09_sampler_interface.html#